src/sidebar/test/oauth-auth-test.js

```diff
@@ -139,22 +69,30 @@ describe('sidebar.oauth-auth', function () {
       }],
     };
 
-    fakeWindow = new FakeWindow();
-
     fakeLocalStorage = {
       getObject: sinon.stub().returns(null),
       setObject: sinon.stub(),
       removeItem: sinon.stub(),
     };
 
+    fakeClient = {
+      exchangeAuthCode: sinon.stub().returns(Promise.resolve(null)),
+      exchangeGrantToken: sinon.stub().returns(Promise.resolve(null)),
+      revokeToken: sinon.stub().returns(Promise.resolve(null)),
+      refreshToken: sinon.stub().returns(Promise.resolve(null)),
+      authorize: sinon.stub().returns(Promise.resolve(null)),
+    };
+
+    fakeWindow = new FakeWindow;
+
     angular.mock.module('app', {
-      $http: fakeHttp,
+      $http: {},
       $window: fakeWindow,
       apiRoutes: fakeApiRoutes,
       flash: fakeFlash,
       localStorage: fakeLocalStorage,
-      random: fakeRandom,
       settings: fakeSettings,
+      OAuthClient: () => fakeClient,
```

Comment:
An unusual feature of JavaScript constructor functions is that they can _return_ an object which becomes the _result_ of the `new FooClass` call. This provides a convenient way to mock classes (rather than instances) in tests.

We actually do this implicitly all over the place in the code (Angular "services" are essentially class constructors).